### PR TITLE
Apply similar changes as #596 to the internal data

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: internal/data/metric.go [hunk 1]
```diff
@@ -599,33 +639,36 @@ type SummaryValueAtPercentile struct {
...
 	return wrappers
 }
 
-func (vp *SummaryValueAtPercentile) Percentile() float64 {
+func (vp SummaryValueAtPercentile) Percentile() float64 {
 	return vp.orig.Percentile
 }
 
-func (vp *SummaryValueAtPercentile) SetPercentile(v float64) {
+func (vp SummaryValueAtPercentile) SetPercentile(v float64) {
```
Comment (rghetia) [inline internal/data/metric.go]: Without pointer receiver set is not meaningful.

Comment (bogdandrutu) [reply internal/data/metric.go]: it is because of pimpl

Comment (rghetia) [reply internal/data/metric.go]: sorry, my bad. it is not an issue.

**Comment by tigrannajaryan** [comment]
Is this still needed since you now have https://github.com/open-telemetry/opentelemetry-collector/pull/624 ?

**Comment by tigrannajaryan** [comment]
Closing this, I don't think this is needed anymore.

**Pull request closed without merging**